gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -119,80 +116,95 @@ private static void configureSourceSet(
+                        // it
+                        // directly because that would require us to depend on Gradle 5.6.
+                        // Instead, we just copy the attributes from compileClasspath.
+                        compileClasspath.get().getAttributes().keySet().forEach(attribute -> {
+                            Object value =
+                                    compileClasspath.get().getAttributes().getAttribute(attribute);
+                            conf.getAttributes().attribute((Attribute<Object>) attribute, value);
+                        });
 
-        // Figure out what our compile dependencies are while ignoring dependencies we've inherited from other source
-        // sets. For example, if we are `test`, some of our configurations extend from the `main` source set:
-        // testImplementation     extendsFrom(implementation)
-        //  \-- testCompile       extendsFrom(compile)
-        // We therefore want to look at only the dependencies _directly_ declared in the implementation and compile
-        // configurations (belonging to our source set)
-        project.afterEvaluate(p -> {
-            Configuration implCopy = implementation.copy();
-            // Without these, explicitCompile will successfully resolve 0 files and you'll waste 1 hour trying
-            // to figure out why.
-            project.getConfigurations().add(implCopy);
-
-            explicitCompile.extendsFrom(implCopy);
-
-            // For Gradle 6 and below, the compile configuration might still be used.
-            maybeCompile.ifPresent(compile -> {
-                Configuration compileCopy = compile.copy();
-                // Ensure it's not resolvable, otherwise plugins that resolve all configurations might have
-                // a bad time resolving this with GCV, if you have direct dependencies without corresponding entries in
-                // versions.props, but instead rely on getting a version for them from the lock file.
-                compileCopy.setCanBeResolved(false);
-                compileCopy.setCanBeConsumed(false);
-
-                project.getConfigurations().add(compileCopy);
-
-                explicitCompile.extendsFrom(compileCopy);
-            });
-        });
+                        // Figure out what our compile dependencies are while ignoring dependencies we've inherited from
+                        // other source sets. For example, if we are `test`, some of our configurations extend from the
+                        // `main` source set:
+                        // testImplementation     extendsFrom(implementation)
+                        //  \-- testCompile       extendsFrom(compile)
+                        // We therefore want to look at only the dependencies _directly_ declared in the implementation
+                        // and compile configurations (belonging to our source set)
+                        Configuration implCopy = project.getConfigurations()
```

Comment:
I had originally tried to put this in a `beforeResolve`, where it makes most sense to live. But this causes GCV to have `ConcurrentModificationException`s, as it iterates over all the `Configuration`s and then resolves them, meaning I had to keep it in `afterEvaluate`